consensus/core/src/core_thread.rs

```diff
@@ -102,6 +108,11 @@ impl CoreThread {
                             let _scope = monitored_scope("CoreThread::loop::get_missing");
                             sender.send(self.core.get_missing_blocks()).ok();
                         }
+                        CoreThreadCommand::SetMinProposeRound(round, sender) => {
```

Comment:
Why not use a watch channel for this, and why is the response (sender) needed?

Reply:
We can do a watch channel for sure. I've followed the existing commands approach as that was before you introduced the watch channels. In that case the sender is needed to ensure to synchronize the ordering of actions:
* set min propose round
* new block

otherwise we risk on trying to trigger a new proposal while the min propose round has not been set

Reply:
The overall direction makes sense for using separate channels and for setting properties yes watch channel makes sense as well.